src/huggingface_hub/serialization/_torch.py

```diff
@@ -214,7 +245,10 @@ def save_torch_state_dict(
     # Save the index (if any)
     if state_dict_split.is_sharded:
         index_path = filename_pattern.format(suffix="") + ".index.json"
-        index = {"metadata": state_dict_split.metadata, "weight_map": state_dict_split.tensor_to_filename}
+        index = {
+            "metadata": {**state_dict_split.metadata, **metadata},
+            "weight_map": state_dict_split.tensor_to_filename,
+        }
```

Comment:
Should there be any sanity check on the additional `metadata` if not already done?

Reply:
`metadata` is at the discretion of the frameworks that will use it (transformers/diffusers/accelerate). In practice, I don't think it'll be much used. In any case, we can't really do sanity check since we are supposed to accept anything that is jsonable.